### PR TITLE
Add ABI generation to CI

### PR DESCRIPTION
**Pull request by shahthepro**
If you made a contract change, make sure to complete the checklist below before merging it in master.

Refer to our [documentation](https://github.com/OriginProtocol/security) for more details about contract security best practices.

Contract change checklist:
  - [ ] Code reviewed by 2 reviewers. 
  - [ ] Copy & paste code review [security checklist](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review-Example.md) below this checklist.
  - [ ] Unit tests pass
  - [ ] Slither tests pass with no warning
  - [ ] Echidna tests pass if PR includes changes to OUSD contract (not automated, run manually on local)

### CODE REVIEW

File: contracts/abi.package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "@origin/defi",
+  "license": "MIT",
+  "homepage": "https://github.com/originprotocol/origin-dollar",
+  "engines": {
+    "node": "16"
+  }
```
Comment (rafaelugolini) [inline contracts/abi.package.json:7]: this file is missing a version, and it's probably better to automate the versioning here

File: .github/workflows/abi.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+      - name: "Generate ABI files"
+        run: (yarn run abi:generate) && (yarn run abi:dist)
+        env:
+          CONTRACT_SIZE: true
+        working-directory: ./contracts
+      
+      - name: "Publish"
+        run: npm publish --access public
+        working-directory: ./contracts/dist
```
Comment (rafaelugolini) [inline .github/workflows/abi.yml:39]:
```
env:
          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN_ORIGIN_DEFI}}
```

File: contracts/abi.package.json (hunk 1 above)
Comment (shahthepro) [reply contracts/abi.package.json:7]: Will look into ways to do it

**Comment by shahthepro** [comment]
Merging this in, since it works with the branch now. Gonna test it with release tags once on master

**Pull request merged**